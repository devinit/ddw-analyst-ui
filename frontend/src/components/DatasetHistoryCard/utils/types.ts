export interface SavedQueryData {
  id: number;
  saved_query_db_table: string;
  operation: number;
  full_query: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
}
