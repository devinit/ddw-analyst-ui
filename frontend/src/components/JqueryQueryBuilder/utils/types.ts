export type JqueryQueryBuilderFilter = {
  condition?: string;
  rules?: (JqueryQueryBuilderFilterComparator | JqueryQueryBuilderFilter)[];
};

export type JqueryQueryBuilderHaving = {
  condition?: string;
  rules?: (JqueryQueryBuilderHavingComparator | JqueryQueryBuilderHaving)[];
};

export type JqueryQueryBuilderHavingComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value?: string | number | { plain: string | number } | { column: string; aggregate: string };
  values?: { value: string; label: string }[];
};

export type JqueryQueryBuilderFilterComparator = {
  id: string;
  field: string;
  type: string;
  input: string;
  operator: JqueryQueryBuilderComps;
  value: string | number;
};
export type JqueryQueryBuilderComps =
  | 'equal'
  | 'not_equal'
  | 'less'
  | 'less_or_equal'
  | 'greater'
  | 'greater_or_equal'
  | 'between'
  | 'in';

export type JqueryQueryBuilderFieldData = {
  id: string | number;
  label: string;
  type: string;
};

export type JqueryQueryBuilderIcons = {
  add_group: string;
  add_rule: string;
  remove_group: string;
  remove_rule: string;
  error: string;
};
