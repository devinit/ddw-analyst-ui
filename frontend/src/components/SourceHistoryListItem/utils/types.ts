export interface FrozenData {
  id: number;
  parent_db_table: string;
  frozen_db_table: string | null;
  active: boolean;
  description: string;
  created_on: string;
  user: string;
  status: string;
  logs: string | null;
}
