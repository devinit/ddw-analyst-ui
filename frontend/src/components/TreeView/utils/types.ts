import { ReactElement } from 'react';

export interface TreeViewProps {
  data: Data[];
  depth?: number;

  deleteElement?: ReactElement;

  getStyleClassCb?: (node: Data) => void;

  isCheckable?: (node: Data, depth?: number) => void;
  isDeletable?: (node: Data, depth?: number) => void;
  isExpandable?: (node: Data, depth?: number) => void;

  keywordChildren?: string;
  keywordChildrenLoading?: string;
  keywordKey?: string;
  keywordLabel?: string;

  loadingElement?: ReactElement;
  noChildrenAvailableMessage?: string;

  onCheckToggleCb?: (data: Data[], depth?: number) => void;
  onDeleteCb?: (node: Data, data: Data[], depth?: number) => void;
  onExpandToggleCb?: (node: Data, depth?: number) => void;
  onUpdateCb?: (data: unknown[], depth?: number) => void;

  transitionEnterTimeout?: number;
  transitionExitTimeout?: number;
}

export interface Data {
  id: number;
  name: string;
  isExpanded?: boolean;
  isChecked?: boolean;
  isChildrenLoading?: boolean;
  children?: Data[];
}
