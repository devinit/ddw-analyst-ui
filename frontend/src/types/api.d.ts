export interface APIResponse<T> {
  count: number;
  next: string;
  previous: string;
  results: T;
}
export interface Links {
  next: string | null;
  previous: string | null;
}
export type LinksMap = Map<keyof Links, Links[keyof Links]>;

export interface FetchOptions {
  limit?: number;
  offset?: number;
  link?: string;
  search?: string;
  id?: number;
}
