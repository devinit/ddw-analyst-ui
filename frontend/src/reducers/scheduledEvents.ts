import { List, Map, fromJS } from 'immutable';
import { Action, Reducer } from 'redux';
import { ScheduledEvent, ScheduledEventMap } from '../types/scheduledEvents';

export interface ScheduledEventsAction extends Action {
  scheduledEvents?: ScheduledEvent[];
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeScheduleEventIndex?: number;
  payload: Partial<{
    limit: number;
    offset: number;
    link: string;
    search: string;
    id: number | string;
  }>;
  loading?: boolean;
}
interface State {
  loading: boolean;
  scheduledEvents: List<ScheduledEventMap>;
  activeScheduledEvent?: ScheduledEventMap;
  count: number;
  activeOperationId: number;
}
export type ScheduledEventsState = Map<keyof State, State[keyof State]>;

const prefix = 'scheduledEvents';
export const FETCH_SCHEDULED_EVENTS = `${prefix}.FETCH`;
export const FETCH_SCHEDULED_EVENTS_SUCCESSFUL = `${prefix}.FETCH_SUCCESSFUL`;
export const FETCH_SCHEDULED_EVENTS_FAILED = `${prefix}.FETCH_FAILED`;
export const SET_ACTIVE_SCHEDULED_EVENT = `${prefix}.SET_ACTIVE_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT = `${prefix}.FETCH_SCHEDULED_EVENT`;
export const FETCH_SCHEDULED_EVENT_FAILED = `${prefix}.FETCH_FAILED`;
export const DELETE_SCHEDULED_EVENT = `${prefix}.DELETE_SCHEDULED_EVENT`;
export const DELETE_SCHEDULED_EVENT_SUCCESSFUL = `${prefix}.DELETE_SCHEDULED_EVENT_SUCCESSFUL`;
export const DELETE_SCHEDULED_EVENT_FAILED = `${prefix}.DELETE_SCHEDULED_EVENT_FAILED`;

export const defaultState: ScheduledEventsState = fromJS({
  loading: false,
  scheduledEvents: [],
  activeScheduledEventId: 1,
  limit: 10,
  offset: 0,
});

export const scheduledEventsReducer: Reducer<ScheduledEventsState, ScheduledEventsAction> = (
  state = defaultState,
  action,
) => {
  if (action.type === FETCH_SCHEDULED_EVENTS || action.type === FETCH_SCHEDULED_EVENT) {
    return state.set('loading', true);
  }
  if (action.type === FETCH_SCHEDULED_EVENTS_SUCCESSFUL && action.scheduledEvents) {
    return state.withMutations((map) =>
      map
        .set('loading', false)
        .set('scheduledEvents', fromJS(action.scheduledEvents))
        .set('count', action.count),
    );
  }
  if (
    action.type === FETCH_SCHEDULED_EVENTS_FAILED ||
    action.type === FETCH_SCHEDULED_EVENT_FAILED
  ) {
    return state.set('loading', false);
  }
  if (action.type === SET_ACTIVE_SCHEDULED_EVENT) {
    return state.set('activeScheduledEvent', action.activeScheduledEvent).set('loading', false);
  }

  return state;
};
