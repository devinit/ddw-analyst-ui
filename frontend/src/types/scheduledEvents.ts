import { Map } from 'immutable';

export interface ScheduledEvent {
  id: number;
  name: string;
  description: string | null;
  enabled: boolean;
  interval: number;
  interval_type: string;
  repeat: string;
}

export type ScheduledEventMap = Map<keyof ScheduledEvent, ScheduledEvent[keyof ScheduledEvent]>;
