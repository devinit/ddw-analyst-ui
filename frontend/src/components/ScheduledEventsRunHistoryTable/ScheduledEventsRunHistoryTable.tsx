import React, { FunctionComponent, ReactNode, useState } from 'react';
import { Table } from 'react-bootstrap';
import { ScheduledEventRunHistory } from '../../types/scheduledEvents';
import { BasicModal } from '../BasicModal';
import { ScheduledEventsRunHistoryTableRow } from '../ScheduledEventsRunHistoryTableRow';

interface ScheduledEventRunHistoryTableProps {
  data: ScheduledEventRunHistory[];
}
export const ScheduledEventsRunHistoryTable: FunctionComponent<ScheduledEventRunHistoryTableProps> = (
  props,
) => {
  const [info, setInfo] = useState('');
  const onViewLogs = (logs: string): void => setInfo(logs);
  const onModalHide = () => setInfo('');
  const renderRows = (): ReactNode =>
    props.data
      ? props.data.map((history: ScheduledEventRunHistory, index: number) => (
          <ScheduledEventsRunHistoryTableRow
            key={index}
            history={history}
            onViewLogs={onViewLogs}
          />
        ))
      : null;

  return (
    <Table responsive striped>
      <thead>
        <tr>
          <th>Status</th>
          <th>Started</th>
          <th>Ended</th>
          <th>Actions</th>
        </tr>
      </thead>
      <tbody>{renderRows()}</tbody>
      <BasicModal show={!!info} onHide={onModalHide}>
        <p>{info}</p>
      </BasicModal>
    </Table>
  );
};
