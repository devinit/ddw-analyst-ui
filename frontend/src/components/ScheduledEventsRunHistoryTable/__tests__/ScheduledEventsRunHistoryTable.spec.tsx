/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { ScheduledEventRunHistory } from '../../../types/scheduledEvents';
import { ScheduledEventsRunHistoryTable } from '../ScheduledEventsRunHistoryTable';

describe('<ScheduledEventsRunHistoryTable', () => {
  const data: ScheduledEventRunHistory[] = [
    {
      scheduled_event: 1,
      start_at: '',
      status: 'pending',
    },
  ];
  afterEach(cleanup);
  const { container } = render(<ScheduledEventsRunHistoryTable data={data} />);
  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
