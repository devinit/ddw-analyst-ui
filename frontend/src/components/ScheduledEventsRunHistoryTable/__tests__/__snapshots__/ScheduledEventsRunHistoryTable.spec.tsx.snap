// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduledEventsRunHistoryTable should match snapshot 1`] = `
<div>
  <div
    class="table-responsive"
  >
    <table
      class="table table-striped"
    >
      <thead>
        <tr>
          <th>
            Status
          </th>
          <th>
            Started
          </th>
          <th>
            Ended
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <span
              class="badge"
            />
          </td>
          <td>
            Invalid date
          </td>
          <td />
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
